python_dependency_verifier/python_dependency_verifier.py

```diff
@@ -137,63 +53,150 @@ def get_latest_version_number(package_name):
-            if field_dependency_name not in file_contents:
-                continue
-            dependency_list = get_dependency_with_latest_version(
-                get_install_requires_field_contents(file_contents),
-                ignore_this)
-            dependency_list += get_append_list(file_contents, ignore_this)
-            if dependency_list is not []:
-                print filename
-                print dependency_list
-
-        except IOError:
-            continue
-
-
-def check_all_filename_in_subdirs(path, filename_to_search_for,
-                                  field_dependency_name, ignore_this):
-    """receives a path and setup filename to search for,
-    a field dependency name and regex to ignore
-    returns a dict of filename and the list of dependencies it uses,
-    whether they are version locked,
-    which version and the latest version of the package available on pip
-    """
-    result = []
-    for dir in os.listdir(path):
-        try:
-            filename = path + dir + filename_to_search_for
-            file_contents = get_file_contents(filename)
-            file_contents = file_contents.replace(" = ", "=")
-            if field_dependency_name not in file_contents:
-                continue
-            dependency_list = get_dependency_with_latest_version(
-                get_install_requires_field_contents(file_contents),
-                ignore_this)
-            dependency_list += get_append_list(file_contents, ignore_this)
-            if dependency_list is not []:
-                result.append([filename, dependency_list])
-
-        except IOError:
-            continue
-
-    return result
+def fix_unequal_conditions(string):
+    """makes <> conditions human readable"""
+    return string.replace(",<", "<")
+
+
+def remove_whitespace_from_equals(string):
+    """receives string and returns it without inline comments"""
+    return string.replace(" =", "=").replace("= ", "=")
+
+
+def remove_comments(string):
+    """receives string and returns it without inline comments"""
+    return re.sub(re.compile("#.*?\n"), "", string)
+
+
+def get_file_contents(path_to_file):
+    """receives a path to file and returns its contents as string"""
+    contents = None
+    try:
+        with open(path_to_file, 'r') as f:
+            contents = f.read()
+    except IOError as ex:
+        logging.error("unable to open file {0}".format(path_to_file))
+        logging.error(ex)
+        return ''
+    return contents
+
+
+def filter_list_for_regex(list_of_strings, filter_regex):
+    return [string for string in list_of_strings
+            if not re.match(re.compile(filter_regex), string)]
+
+def remove_quotes_and_whitespace_from_list(list):
+    return [remove_quotes_and_whitespace(x) for x in list]
+
+
+def remove_quotes_and_whitespace(string):
+    return string.replace(" ", "").replace(
+        "\'", "").replace("\"", "")
+
+
+class PythonSetuptoolsDependencyCheckerForFile():
```

Comment:
Well then, it will make more sense to create a class that handles a file and then have to functions outside of the class, one to instantiate the class once and process a file and one to call this function iteratively.